src/lib.rs

```diff
@@ -1,20 +1,68 @@
+#![doc = include_str!("../README.md")]
+#[doc(hidden)]
 pub mod attr;
+#[doc(hidden)]
 pub mod pretty;
+#[doc(hidden)]
 pub mod util;
 
+#[doc(hidden)]
 pub use attr::calculate_attributes;
+#[doc(hidden)]
 pub use pretty::PrettyPrinter;
 
-pub fn pretty_print(content: &str, width: usize) -> String {
-    let root = typst_syntax::parse(content);
-    if root.erroneous() {
-        return content.to_string();
+use typst_syntax::{Source, SyntaxNode};
+
+/// Entry point for pretty printing a typst document.
+#[derive(Debug, Clone)]
+pub struct Typstyle {
+    content: String,
+    root: SyntaxNode,
+    width: usize,
+}
+
+impl Typstyle {
+    /// Create a new Typstyle instance from a string.
+    /// # Example
+    /// ```rust
+    /// use typstyle_core::Typstyle;
+    /// let content = "#{1+1}";
+    /// let res = Typstyle::with_content(content.to_string(), 80).pretty_print();
+    /// assert_eq!(res, "#{\n  1 + 1\n}");
+    /// ```
+    pub fn with_content(content: String, width: usize) -> Self {
```

Comment:
A bikeshedding, The `with_xxx` functions should have signature like `fn with_xxx(self, ...) -> Self`, so the naming `with_content` and `with_src` are confusing here. It would like to be renamed:
+ `with_content` -> `new_content`
+ `with_src` -> `new_src`

Reply:
https://rust-lang.github.io/api-guidelines/predictability.html#constructors-are-static-inherent-methods-c-ctor


maybe new_with_src, according to guideline